manifests/database.pp

```diff
@@ -8,12 +8,34 @@
     Service['mongodb'] -> Exec['migrate_pulp_db']
   }
 
+  transition { 'stop pulp_workers':
```

Comment:
How about:
```puppet
['workers', 'streamer', 'celerybeat', 'resource_manager'].each |$service| {
  transition { "stop pulp_${service}":
    resource   => Service["pulp_${service}"],
    attributes => { ensure => 'stopped' },
    prior_to   => Exec['migrate_pulp_db'],
  }

  Exec['migrate_pulp_db'] ~> Service["pulp_${service}"]
}
```

I'm not entirely sure the refresh is needed since init.pp has it, but I left it in the example here just to show how it can be done with less code duplication.
https://github.com/theforeman/puppet-pulp/blob/1300985bfa981738afbd9f9895581d8753d65b80/manifests/init.pp#L573